### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by aseure**

### CODE REVIEW

File: algoliasearch/index.go [hunk 1]
```diff
@@ -54,29 +66,41 @@ func (i *Index) GetObjects(objectIDs ...string) (interface{}, error) {
...
 func (i *Index) DeleteObject(objectID string) (interface{}, error) {
 	return i.client.transport.request("DELETE", "/1/indexes/"+i.nameEncoded+"/"+i.client.transport.urlEncode(objectID), nil, write)
 }
 
+// GetSettings retrieves the index settings.
 func (i *Index) GetSettings() (interface{}, error) {
 	return i.client.transport.request("GET", "/1/indexes/"+i.nameEncoded+"/settings", nil, read)
 }
 
+// GetSettings retrieves the index settings.
```
Comment (ercolanelli-leo) [inline algoliasearch/index.go]: Copy-paste artifact

File: algoliasearch/index.go [hunk 2]
```diff
@@ -202,17 +250,21 @@ func (i *Index) Batch(objects interface{}, actions []string) (interface{}, error
...
+// CustomBatch actually performs the batch request of all `queries`.
 func (i *Index) CustomBatch(queries interface{}) (interface{}, error) {
 	request := make(map[string]interface{})
 	request["requests"] = queries
 	return i.client.transport.request("POST", "/1/indexes/"+i.nameEncoded+"/batch", request, write)
 }
 
-// Deprecated use BrowseFrom or BrowseAll
+// Browse returns `hitsPerPage` results from the `page` page.
+// [DEPRECATED] Use `BrowseFrom` or `BrowseAll` instead.
```
Comment (ercolanelli-leo) [inline algoliasearch/index.go]: From [official godoc documentation](https://blog.golang.org/godoc-documenting-go-code):
> To signal that an identifier should not be used, add a paragraph to its doc comment that begins with "Deprecated:" followed by some information about the deprecation.

File: algoliasearch/index.go [hunk 3]
```diff
@@ -283,6 +356,9 @@ func (i *Index) UpdateKey(key string, acl []string, validity int, maxQueriesPerI
 	return i.UpdateKeyWithParam(key, body)
 }
 
+// UpdateKeyWithParam updates the `key` API key according to the `params`
+// parameters which is a `map[string]interface{}` of all the parameters given
```
Comment (ercolanelli-leo) [inline algoliasearch/index.go]: Again :cry:

File: algoliasearch/index.go [hunk 4]
```diff
@@ -270,10 +339,14 @@ func (i *Index) AddKey(acl []string, validity int, maxQueriesPerIPPerHour int, m
 	return i.AddKeyWithParam(body)
 }
 
+// AddKeyWithParam registers a new API for the index. The `params` parameter is
+// a `map[string]interface{}` of all the parameters given to the `AddKey`
```
Comment (ercolanelli-leo) [inline algoliasearch/index.go]: Makes me wanna cry... (but nothing we can do right now)

File: algoliasearch/index.go (hunk 2 above)
Comment (aseure) [inline algoliasearch/index.go]: Thank you, I didn't know that.

**Pull request merged**